sql/catalyst/src/main/scala/org/apache/spark/sql/Row.scala

```diff
@@ -59,8 +59,18 @@ object Row {
    */
   @deprecated("This method is deprecated and will be removed in future versions.", "3.0.0")
   def merge(rows: Row*): Row = {
-    // TODO: Improve the performance of this if used in performance critical part.
-    new GenericRow(rows.flatMap(_.toSeq).toArray)
+    var elementNumber = 0
+    for (row <- rows)
+      elementNumber += row.length
+    val allValues = new Array[Any](elementNumber)
+    var arrayPos = 0
+    for (row <- rows) {
+      for (rowIndex <- (0 until row.length)) {
+        allValues.update(arrayPos, row.get(rowIndex))
+        arrayPos += 1
+      }
+    }
+    new GenericRow(allValues)
```

Comment:
Hi, @g1thubhub . Thank you for showing your interest to improve Apache Spark, but this function is deprecated and will be removed. So, there is no need to improve this at this time. Could you close this PR?